var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
@tgamblin We already have what you need:
```python
@AutotoolsPackage.sanity_check('configure')
def ubuntu_fix(self):
   spec = self.spec
   if '@:1.8.14' in spec:
       ...
```
I will submit next week a PR to alias the `precondition` decorator to `run_before` and `sanity_check` to `run_after` (as @davydden suggested a while ago), so we can give the correct semantic sense to each function, even though the behavior is the same.